include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -2806,6 +2806,16 @@ AffineConstraints<number>::distribute(VectorType &vec) const
 {
   Assert(sorted == true, ExcMatrixNotClosed());
 
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+  if constexpr (std::is_same<
+                  VectorType,
+                  LinearAlgebra::TpetraWrappers::Vector<number>>::value)
+    Assert(vec.is_compressed(),
+           ExcMessage("To distribute a parallel vector, the vector must "
+                      "not contain ghost elements. Therefore, compress() "
+                      "must be called on the vector first."));
```

Comment:
Adding the assertion for a specific vector class is fine by me, but I think the text of the error message is not quite on point. Having ghost elements (as opposed to "fully distributed") is orthogonal to having called `compress()`. In fact, you can't even call `compress()` on ghosted vectors because these vectors are read-only anyway. So perhaps the following would be better?
```suggestion
    Assert(vec.is_compressed(),
           ExcMessage("To distribute a parallel vector, the vector must "
                      "not contain any previously written-to or added-to entries that "
                      "have not already been sent to these entries' owners. But "
                      "the vector you passed in has such entries. You need to call "
                      "compress() on the vector first, before passing it to this function."));
```

Reply:
Thank you for the feedback! Yes, that describes the setting better.